tests/flux-v2-image-automation.sh

```diff
@@ -56,7 +62,58 @@ for FILE_LOCATION in $(echo ${FILE_LOCATIONS}); do
                 echo "Non whitelisted pattern found in ImagePolicy: $IMAGE_POLICY it should be ^prod-[a-f0-9]+-(?P<ts>[0-9]+)" && exit 1
             fi
         done
+    done
+
+    ##############################################################################################################
+    # Print success if ALL image policies are clean
+    ##############################################################################################################
+    printf "\n\n########## Image Policy documents checked and passing ########## \n\n"
+
+    ##############################################################################################################
+    # This section compiles a list of files that contains `image:` and stores them in an array HELMRELEASES
+    # Only files that follow these rules will be added to the array:
+    # - Contains the `image:` string
+    # - Is HelmRelease type document
+    ##############################################################################################################
+
+    HELMRELEASES=()
+    for FILE in $(grep -lr "image:" $FILE_LOCATION | grep -Ev "$EXCLUSIONS" ); do
+        while read -r doc; do
+            if [ "$doc" == "HelmRelease" ]; then
+                IFS=$'\n'
+                HELMRELEASES+=("$FILE")
+            fi
+        done < <(yq eval '.kind' $FILE)
+    done
+
+    OUTPUTFILE="images.yaml"
+    DIRECTORIES=$FILE_LOCATIONS
+
+    for dir in $DIRECTORIES; do
+        exclude=false
+        for EXCLUDED_PATH in "${HELMRELEASES[@]}"; do
+            if [[ $dir == $EXCLUDED_PATH* ]]; then
+                exclude=true
+                break
+            fi
+        done
 
+        if ! $exclude && ls "$dir" | grep -q -E 'kustomization\.ya?ml'; then
+            kustomize build --load-restrictor LoadRestrictionsNone "$dir" 2>&1 | yq eval 'select(.kind == "HelmRelease" and (.spec.values.image or .spec.values.*.image != null))' >> $OUTPUTFILE
```

Comment:
You're using kustomize to find HelmRelease documents which is what the while loop on line 81 already does.
Do every file inside the HELMRELEASES array is `kind: HelmRelease`

Not using the OUTPUTFILE after this point so whats the benefit of saving it to file?

Reply:
I think the idea run this   then use yq to select the image on it          kustomize build --load-restrictor LoadRestrictionsNone "$dir" 2>&1 | yq eval 'select(.kind == "HelmRelease" and (.spec.values.image or .spec.values.*.image != null))' >> $OUTPUTFILE